### PR TITLE
Plane: Refactor plane failsafe

### PR DESCRIPTION
**Pull request by magicrub**
Refactor of Plane::Advanced_Failsafe to use it's own file. This architecture brings it inline to match rover and copter.

Also added #if ADVENCED_FAILSAFE == ENABLED everywhere.

compiled testing with enabled and disabled.

I've done this in two commits. One that is just the header refactor and the other is the #if ADVENCED_FAILSAFE sprinkled everywhere. We should definitely do the refactor, but I consider the #if stuff as an optional merge

### CODE REVIEW

File: ArduPlane/events.cpp [hunk 1]
```diff
@@ -169,9 +169,13 @@ void Plane::handle_battery_failsafe(const char *type_str, const int8_t action)
             break;
 
         case Failsafe_Action_Terminate:
+#if ADVANCED_FAILSAFE == ENABLED
             char battery_type_str[17];
             snprintf(battery_type_str, 17, "%s battery", type_str);
             afs.gcs_terminate(true, battery_type_str);
+#else
+            disarm_motors();
```
Comment (WickedShell) [inline ArduPlane/events.cpp]: This is wrong IMO. Disarming motors is quite different then terminating the vehicle. If AFS isn't available you shouldn't be able to terminate. (Although arguably we have a preexisting bug as we didn't respect the return value on `gcs_terminate` anyways)

Comment (magicrub) [reply ArduPlane/events.cpp]: Then we also have a different bug in the other vehicles because this matches their behavior

Comment (WickedShell) [reply ArduPlane/events.cpp]: On other vehicles disarming motors is actually symmetric with terminating. On plane you will continue to glide away gently into the ground, which can be a very very large distance that the vehicle travels. (That being said a disarm only failsafe may be desirable). Terminate should minimize any and all movement. On a copter stopping motors is termination :)

Comment (magicrub) [reply ArduPlane/events.cpp]: So... How about we reverse all flight surfaces except throttle?

**Pull request merged**